commands/databases.go

```diff
@@ -0,0 +1,851 @@
+	CmdBuilder(cmd, RunDatabaseUserGet, "get <database-id> <user-id>", "get a database user", Writer, aliasOpt("g"))
+	CmdBuilder(cmd, RunDatabaseUserCreate, "create <database-id> <user-name>", "create a database user", Writer, aliasOpt("c"))
+
+	cmdDatabaseUserDelete := CmdBuilder(cmd, RunDatabaseUserDelete, "delete <database-id> <user-id>", "delete database cluster", Writer,
+		aliasOpt("rm"))
+	AddBoolFlag(cmdDatabaseUserDelete, doctl.ArgForce, doctl.ArgShortForce, false, "force database delete")
+
+	return cmd
+}
+
+// Database Users
+
+// RunDatabaseUserList retrieves a list of users for specific database cluster
+func RunDatabaseUserList(c *CmdConfig) error {
+	if len(c.Args) == 0 {
+		return doctl.NewMissingArgsErr(c.NS)
+	}
+
+	id := c.Args[0]
+
+	users, err := c.Databases().ListUsers(id)
+	if err != nil {
+		return err
+	}
+
+	return displayDatabaseUsers(c, users...)
+}
+
+// RunDatabaseUserGet retrieves a database user for a specific database cluster
+func RunDatabaseUserGet(c *CmdConfig) error {
+	if len(c.Args) < 2 {
+		return doctl.NewMissingArgsErr(c.NS)
+	}
+
+	databaseID := c.Args[0]
+	userID := c.Args[1]
+
+	user, err := c.Databases().GetUser(databaseID, userID)
+	if err != nil {
+		return err
+	}
+
+	return displayDatabaseUsers(c, *user)
+}
+
+// RunDatabaseUserCreate creates a database user for a database cluster
+func RunDatabaseUserCreate(c *CmdConfig) error {
+	if len(c.Args) < 2 {
+		return doctl.NewMissingArgsErr(c.NS)
+	}
+
+	databaseID := c.Args[0]
+	req := &godo.DatabaseCreateUserRequest{Name: c.Args[1]}
+
+	user, err := c.Databases().CreateUser(databaseID, req)
+	if err != nil {
+		return err
+	}
+
+	return displayDatabaseUsers(c, *user)
+}
+
+// RunDatabaseUserDelete deletes a database user
+func RunDatabaseUserDelete(c *CmdConfig) error {
+	if len(c.Args) < 2 {
+		return doctl.NewMissingArgsErr(c.NS)
+	}
+
+	force, err := c.Doit.GetBool(c.NS, doctl.ArgForce)
+	if err != nil {
+		return err
+	}
+
+	if force || AskForConfirm("delete this database user") == nil {
+		databaseID := c.Args[0]
+		userID := c.Args[1]
+		return c.Databases().DeleteUser(databaseID, userID)
+	}
+
+	return fmt.Errorf("operation aborted")
+}
+
+func displayDatabaseUsers(c *CmdConfig, users ...do.DatabaseUser) error {
+	item := &displayers.DatabaseUsers{DatabaseUsers: users}
+	return c.Display(item)
+}
+
+func databasePool() *Command {
+	cmd := &Command{
+		Command: &cobra.Command{
+			Use:     "pool",
+			Aliases: []string{"p"},
+			Short:   "database pool commands",
+			Long:    "database is used to access database pool commands",
+		},
+	}
+
+	CmdBuilder(cmd, RunDatabasePoolList, "list <database-id>", "list database pools", Writer, aliasOpt("ls"))
+	CmdBuilder(cmd, RunDatabasePoolGet, "get <database-id> <pool-id>", "get a database pool", Writer, aliasOpt("g"))
+	cmdDatabasePoolCreate := CmdBuilder(cmd, RunDatabasePoolCreate, "create <database-id> <pool-name>", "create a database pool", Writer,
```

Comment:
Creating pools doesn't seem to work for me. The API is returning a 422 `invalid pool mode`, but I think that is the actual issue.  

```
l$ ./builds/doctl databases pool create 28ddd061-0822-47bd-9c12-d2f46747cc6c backend --db defaultdb --size 10 --user doadmin --trace
doctl: 2019/04/03 15:18:49 -> "POST /v2/databases/28ddd061-0822-47bd-9c12-d2f46747cc6c/pools HTTP/1.1\r\nHost: api.digitalocean.com\r\nUser-Agent: doctl/1.15.0-dev godo/1.10.1\r\nContent-Length: 111\r\nAccept: application/json\r\nContent-Type: application/json\r\nAccept-Encoding: gzip\r\n\r\n{\"pool\":{\"user\":\"doadmin\",\"name\":\"backend\",\"size\":10,\"db\":\"defaultdb\",\"mode\":\"transaction\",\"connection\":null}}\n"
create a database pooldoctl: 2019/04/03 15:18:54 <- "HTTP/2.0 422 Unprocessable Entity\r\nContent-Length: 112\r\nCf-Ray: 4c1d641fbe23c5c2-EWR\r\nContent-Type: text/plain; charset=utf-8\r\nDate: Wed, 03 Apr 2019 19:18:54 GMT\r\nExpect-Ct: max-age=604800, report-uri=\"https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct\"\r\nRatelimit-Limit: 20000\r\nRatelimit-Remaining: 19973\r\nRatelimit-Reset: 1554316089\r\nServer: cloudflare\r\nSet-Cookie: __cfduid=d4464443c15e3ae84240c94ba43953b0e1554319134; expires=Thu, 02-Apr-20 19:18:54 GMT; path=/; domain=.digitalocean.com; HttpOnly\r\nX-Gateway: Edge-Gateway\r\nX-Request-Id: 2795f022-643c-49cb-b774-3d950b522224\r\nX-Response-From: service\r\n\r\n{\"message\":\"invalid pool mode\",\"id\":\"unprocessable_entity\",\"request_id\":\"2795f022-643c-49cb-b774-3d950b522224\"}\n"

# [snipping usage info]

Error: POST https://api.digitalocean.com/v2/databases/28ddd061-0822-47bd-9c12-d2f46747cc6c/pools: 422 (request "2795f022-643c-49cb-b774-3d950b522224") invalid pool mode
```

Reply:
Hm, will have to look at that

Reply:
Good catch, it was a bug in `godo` that has been fixed. I've tested this and it works for me